### PR TITLE
Glasgow 6 | Haroun Alarabi | Full-Stack-Project-Assessment | Level 300

### PR DESCRIPTION
**Pull request by HarounAlarabi**
https://verdant-crisp-d6d210.netlify.app/
Hi @Dedekind561 I'm attempting to display data from Render in browser but i's not showing up. However, when I use Postman to retrieve the data from Render, it works as expected."

### CODE REVIEW

**Comment by Dedekind561** [comment]
Hi @HarounAlarabi , 
Thankyou for submitting Level 300 of the Full Stack Project Assessment. However, at the moment, I can't review your work properly. Locally I'm just seeing a blank screen with a heading **Video Recommendation** - do you have a deployed version of the site itself?

**Comment by Dedekind561** [comment]
Morning Haroun,
Thankyou for submitting a link to your deployed application.
At the moment, I can't review your work because you've committed the `node_modules`. Please can you remove them from your pull request then I can take a look :)

**Comment by Dedekind561** [comment]
The deployed version of your app is now working for me at the moment :)

**Comment by HarounAlarabi** [comment]
Hi @Dedekind561  I have commited the code again with changes

File: client/src/components/AddVideo.jsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+      console.log("Original URL:", url);
+      console.log("Embed URL:", embedUrl);
+
+      const newInput = {
+        title: title,
+        url: embedUrl,
+        rating: 0,
+      };
+
+      onVidAdd(newInput);
```
Comment (Dedekind561) [inline client/src/components/AddVideo.jsx:28]: Unclear to me what this function does just by reading its name `onVidAdd`

File: client/src/components/VideosList.jsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+import "../index.css";
+import Search from "./Search";
+
+function VideosList() {
+  const [videos, setVideos] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [searchResult, setSearchResult] = useState([]);
+
+  useEffect(() => {
+    fetch("https://node-server-full-stack.onrender.com/videos")
```
Comment (Dedekind561) [inline client/src/components/VideosList.jsx:13]: Use an environment variable for this API url - don't just hardcode the string. Do you understand why this is important?

File: server/DBConfig.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+const { Pool } = require("pg");
+const videosPool = new Pool({
+  connectionString: "postgres://full_stack_database_jp62_user:JfOA8mzl7maxtJEnTL28nmslJ5JSqVfe@dpg-ck2p2hvqj8ts73fm2m5g-a.oregon-postgres.render.com/full_stack_database_jp62",
```
Comment (Dedekind561) [inline server/DBConfig.js:3]: ❗ You've just made your database connection string public! You need to store strings like this in env variables 👀

File: client/src/index.css [hunk 4]
```diff
@@ -11,3 +11,43 @@ code {
   font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
     monospace;
 }
+
+form{
+margin: 20px;
+max-width: 400px;
```
Comment (Dedekind561) [inline client/src/index.css:17]: This CSS rule isn't indented properly: how can you fix this 🔨 ?

File: client/src/components/VideosList.jsx [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+      setVideos(videoDeleted);
+  })
+  };
+
+  const fetchVideos = () => {
+    fetch("https://node-server-full-stack.onrender.com/videos")
+      .then((res) => res.json())
+      .then((data) => {
+        setVideos(data);
+        searchResult(data);
```
Comment (Dedekind561) [inline client/src/components/VideosList.jsx:70]: Here you're fetching the data and then you
1. `call res.json()`
2. call `setVideos` with the data
3. and then call `searchResult` -> can you explain the issue here?

File: client/src/components/VideosList.jsx [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+    newInput.rating = 0;
+    fetch("https://node-server-full-stack.onrender.com/videos", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/JSON",
+      },
+      body: JSON.stringify(newInput),
+    })
+      .then((res) => {
+        fetchVideos();
```
Comment (Dedekind561) [inline client/src/components/VideosList.jsx:36]: Call to `fetchVideos` here - describe the effect of calling this function here

File: client/src/components/VideosList.jsx [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+        return res.json();
+      })
+      .then((data) => {
+        if (data && data.message === "Video stored successfully") {
```
Comment (Dedekind561) [inline client/src/components/VideosList.jsx:40]: Now you're calling `setVideos` down here as well - I think you need to think very carefully about what happens when after you've POSTed a new video to your database

**Review by Dedekind561 — COMMENTED** [review]
Hi Haroun, 
Thanks for your submission,
I've just taken a look at your code.
The deployed application has the core functionality; however,  you need to look more carefully at your logic when adding a new video. Think carefully how the state of your application updates after you've made the POST request. Write down the steps on your piece of paper and think about it carefully to see if you can identify the issues.

Well done on your work so far though 🎉

File: client/src/components/VideosList.jsx (hunk 2 above)
Comment (HarounAlarabi) [reply client/src/components/VideosList.jsx:13]: I chaned this

File: server/DBConfig.js (hunk 3 above)
Comment (HarounAlarabi) [reply server/DBConfig.js:3]: I have tried before to use variables to store it but I got an error then I used this way

File: client/src/components/VideosList.jsx (hunk 5 above)
Comment (HarounAlarabi) [reply client/src/components/VideosList.jsx:70]: the issue here is I used searchResult , so I have to change in to setSearchResult()

File: client/src/components/VideosList.jsx (hunk 6 above)
Comment (HarounAlarabi) [reply client/src/components/VideosList.jsx:36]: fetchVideos to update the list of videos after adding new video

**Pull request closed without merging**